project/src/main/java/com/google/sps/servlets/RecommendationServlet.java

```diff
@@ -26,35 +28,49 @@
 public class RecommendationServlet extends HttpServlet {
   private final Gson gson = new Gson();
   private static final Logger LOGGER = Logger.getLogger(RecommendationServlet.class.getName());
+  private static Set<Restaurant> restaurantSet;
+  private static Map<Restaurant, Double> restaurantScores;
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Set<Restaurant> restaurantSet = new HashSet<>();
+    restaurantSet = new HashSet<>();
+    restaurantScores = new HashMap<>();
     BufferedReader reader = request.getReader();
     String body = reader.readLine();
     try {
-      addRestaurantsToSet(body, restaurantSet);
+      JSONObject reqBody = new JSONObject(body);
+      JSONArray restaurantList = reqBody.getJSONArray("restaurants");
+      addRestaurantsToSet(restaurantList);
+      JSONObject preferences = reqBody.getJSONObject("preferences");
+      scoreRestaurants(preferences);
```

Comment:
and you can probably omit some local vars here as I think this is is clear enough even written as 
scoreRestaurants(reqBody.getJSONArray("restaurants"), reqBody.getJsonObject("preferences"));